scipy/interpolate/_bsplines.py

```diff
@@ -1367,7 +1367,7 @@ def _make_periodic_spline(x, y, t, k, axis):
 
 
 def make_interp_spline(x, y, k=3, t=None, bc_type=None, axis=0,
-                       check_finite=True):
+                       check_finite=True, periodic=False):
```

Comment:
Please reuse the already-existing `bc_type={"periodic", "not-a-knot"}` API. This relates to all public functions: make_splrep, make_splprep (is it a part of this first PR?) and generate_knots. There is machinery for validating the `bc_type` argument in `_bsplines.py`, feel free to reuse. No need to worry about other values of bc_type: just literals, "not-a-knot" (default) and "periodic". 

Private functions can of course keep using whatever is convenient, `periodic={True,False}`.